### DISCUSSION TITLE
having problem with sub-state in editor

### DISCUSSION THREAD

**Discussion by vuguzum** [imjp94/gd-YAFSM#16] [Q&A]
Have found a way to get into the nested state in editor by the source code only )) by d-clicking Not on the label. 
Heh, good work anyway!

**Comment by imjp94** [comment]
That's weird, normally you can just double click on any state node to create nested state machine. Maybe try double click on empty area of state node, instead of the text as that would trigger renaming of node.

If the issue persists, do you mind taking a screenshot?

**Comment by vuguzum** [comment]
Maybe a slight change would work without aiming off label borders:

Click on Unselected state rect -> select state rect
Click on Selected state rect -> edit label
Double click state rect -> sublevel

**Reply by imjp94** [reply]
Agree, the UX of opening nested layer is pretty bad. But I felt that the suggested approach is a bit unintuitive.

Anyway, I have just opened a issue(#17 ) for the ease of tracking, and feel free to leave any comments if you have any idea.

For now, the best I can do is minimize the size of the label:
Before
![Screenshot 2021-04-17 235656](https://user-images.githubusercontent.com/11486079/115119351-7f28cf80-9fda-11eb-8e03-94cab989f307.png)
After
![Screenshot 2021-04-17 235724](https://user-images.githubusercontent.com/11486079/115119352-8059fc80-9fda-11eb-9752-839f09fb7213.png)
There's still room for improvement, but it should be good enough for a quick fix.
